pkg/helm/actions/get_repos.go

```diff
@@ -0,0 +1,139 @@
+package actions
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"helm.sh/helm/v3/pkg/repo"
+	"io/ioutil"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"net/http"
+	"sigs.k8s.io/yaml"
+	"sync"
+)
+
+var (
+	helmChartRepositoryGVK = schema.GroupVersionResource{
+		Group:    "helm.openshift.io",
+		Version:  "v1beta1",
+		Resource: "helmchartrepositories",
+	}
+)
+
+func prepareAnnotations(url, name string) map[string]string {
+	return map[string]string{
+		"chart_repo_url": url,
+		"repo_name":      name,
+	}
+}
+
+func getRepoContent(repoURL string, caFile string) (*http.Response, error) {
+	mTLSConfig := &tls.Config{
+		CipherSuites: []uint16{
+			tls.TLS_RSA_WITH_RC4_128_SHA,
+			tls.TLS_RSA_WITH_3DES_EDE_CBC_SHA,
+			tls.TLS_RSA_WITH_AES_128_CBC_SHA,
+			tls.TLS_ECDHE_RSA_WITH_RC4_128_SHA,
+			tls.TLS_RSA_WITH_AES_128_CBC_SHA,
+			tls.TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA,
+		},
+	}
+
+	tr := &http.Transport{
+		TLSClientConfig: mTLSConfig,
+	}
+
+	var rootCAs *x509.CertPool
+	if caFile != "" {
+		rootCAs = x509.NewCertPool()
+		certPEM, err := ioutil.ReadFile(caFile)
+		if err != nil {
+			return nil, err
+		}
+		if !rootCAs.AppendCertsFromPEM(certPEM) {
+			return nil, err
+		}
+	} else {
+		rootCAs, _ = x509.SystemCertPool()
+	}
+	mTLSConfig.ClientCAs = rootCAs
+	c := &http.Client{Transport: tr}
```

Comment:
we should use internal `HelmConfig` struct which would return dynamic http client with TLSConfig required.